main.go

```
@@ -1,25 +1,223 @@
+		offset := (page - 1) * limit
+
+		return db.Limit(limit).Offset(offset)
+	}
+}
+
+func OrderBy(field string, ascending bool) func(db *gorm.DB) *gorm.DB {
+	return func(db *gorm.DB) *gorm.DB {
+		// Basic validation to prevent SQL injection
+		// Only allow alphanumeric characters, underscores, and dots for table.column
+		if !internal.IsValidFieldName(field) {
+			db.AddError(eris.Errorf("invalid field name: %s", field))
```

golangci-lint (1.64.8)

Error return value of `db.AddError` is not checked

(errcheck)